src/slurm_plugin/clustermgtd.py

```diff
@@ -964,6 +973,73 @@ def _is_node_in_replacement_valid(self, node, check_node_is_valid):
             return not time_is_expired if check_node_is_valid else time_is_expired
         return False
 
+    def _handle_ice_nodes(self, active_nodes, unhealthy_dynamic_nodes):
```

Comment:
optional: we can start using Python typing information in new functions: https://docs.python.org/3/library/typing.html

Reply:
Used